src/libstore/binary-cache-store.cc

```diff
@@ -509,7 +509,7 @@ std::shared_ptr<std::string> BinaryCacheStore::getBuildLog(const StorePath & pat
 
     debug("fetching build log from binary cache '%s/%s'", getUri(), logPath);
 
-    return getFile(logPath);
+    return std::move(*getFile(logPath));
```

Comment:
Any reason why you can’t return `getFile(logPath)` since it’s already an `optional<string>`?

(also, wouldn’t this throw if the log isn’t available?)

Reply:
Thanks, that was a SNAFU.